openapi3/schema_test.go

```diff
@@ -1028,7 +1028,10 @@ func testType(t *testing.T, example schemaTypeExample) func(*testing.T) {
 		}
 		for _, typ := range example.AllInvalid {
 			schema := baseSchema.WithFormat(typ)
-			err := schema.Validate(context.Background())
+			ctx := withValidationOptions(context.Background(), &validationOptions{
```

Comment:
Hm this needs a public API, surely users will want to be able to pass validation options to the `Validate` methods implemented to other types than `openapi3.T`.

Reply:
addressed